wallet/src/Pos/Wallet/Web/State/Storage.hs

```diff
@@ -720,7 +722,7 @@ removeNextUpdate :: Update ()
 removeNextUpdate = wsReadyUpdates %= drop 1
 
 -- | Reset the whole database to clean state completely. Used only in testing and debugging.
-testReset :: Update ()
+testReset :: Acid.Update WalletStorage ()
```

Comment:
If the "flexible types" PR is not merged, then we'll need to change the types of all these functions to use the `Acid.Update` or `Acid.Query` types instead of the custom types we have. This would be annoying. Additionally, we'd need to make *another* PR which exposes the `Query` and `Update` destructors (as we use the functions in pure code).